pkg/csi/service/vanilla/controller.go

```diff
@@ -487,9 +487,18 @@ func (c *controller) createBlockVolume(ctx context.Context, req *csi.CreateVolum
 			}
 			if len(queryResult.Volumes) > 0 {
 				// Find datastore topology from the retrieved datastoreURL
+				if queryResult.Volumes[0].DatastoreUrl == "" {
+					msg := fmt.Sprintf("could not retrieve datastore of volume: %q", volumeInfo.VolumeID.Id)
+					log.Error(msg)
+					return nil, status.Error(codes.Internal, msg)
```

Comment:
The successful vpxd task(CNS create) in VC will be present for only few mins. If the retry happens after the task is removed in vpxd, would it result in orphan volume?

Reply:
Yes, but at least this will help prevent creating PVs without node affinity rules, which is more critical than orphan volume.

Reply:
Sure. It seems you agree that this PR is not solving the problem completely but solving part of the problem. Btw, which versions of vSphere has this bug? 

Cc @RaunakShah since he is looking into long term orphan volume solution.

Reply:
@divyenpatel can you respond to my question? And is there a corresponding bug on CNS for this?

Reply:
Testing team observed this issue on `vsphere67p04`. 
3 out of 50 PV's missed to show Node affinity details due to this issue on the NFSv4 datastore.
We have bug filed on CNS for this.